includes/UserMailer.php

```diff
@@ -550,126 +583,104 @@ public function notifyOnPageChange( $editor, $title, $timestamp, $summary, $mino
-						**/
 						&& !$watchingUser->getBoolOption( 'unsubscribed' ) )
-						/**
-						* WIKIA CHANGE end
-						**/
 					{
 						$this->compose( $watchingUser );
 					}
 				}
 			}
 		}
+		$this->emailUsersNotifiedOnAllChanges();
+	}
+
+	private function setReplyToAndFromAddresses() {
+		global $wgPasswordSender, $wgPasswordSenderName, $wgNoReplyAddress;
+		global $wgEnotifFromEditor, $wgEnotifRevealEditorAddress;
+
+		# Reveal the page editor's address as REPLY-TO address only if
+		# the user has not opted-out and the option is enabled at the
+		# global configuration level.
+		$adminAddress = new MailAddress( $wgPasswordSender, $wgPasswordSenderName );
+		if ( $wgEnotifRevealEditorAddress
+			&& ( $this->editor->getEmail() != '' )
+			&& $this->editor->getOption( 'enotifrevealaddr' ) )
+		{
+			$editorAddress = new MailAddress( $this->editor );
+			if ( $wgEnotifFromEditor ) {
+				$this->from    = $editorAddress;
+			} else {
+				$this->from    = $adminAddress;
+				$this->replyto = $editorAddress;
+			}
+		} else {
+			$this->from    = $adminAddress;
+			$this->replyto = new MailAddress( $wgNoReplyAddress );
+		}
+	}
 
+	private function emailUsersNotifiedOnAllChanges() {
 		global $wgUsersNotifiedOnAllChanges;
+
 		foreach ( $wgUsersNotifiedOnAllChanges as $name ) {
-			if ( $editor->getName() == $name ) {
-				// No point notifying the user that actually made the change!
+			// No point notifying the user that actually made the change!
+			if ( $this->editor->getName() == $name ) {
 				continue;
 			}
 			$user = User::newFromName( $name );
 			$this->compose( $user );
 		}
-
-		$this->sendMails();
-		wfProfileOut( __METHOD__ );
 	}
 
 	/**
-	 * @param $editor User
-	 * @param $title Title bool
-	 * @param $minorEdit
 	 * @return bool
 	 */
-	private function canSendUserTalkEmail( $editor, $title, $minorEdit ) {
-		global $wgEnotifUserTalk;
-		$isUserTalkPage = ( $title->getNamespace() == NS_USER_TALK );
-
-		if ( $wgEnotifUserTalk && $isUserTalkPage ) {
-			$targetUser = User::newFromName( $title->getText() );
-
-			if ( !$targetUser || $targetUser->isAnon() ) {
-				wfDebug( __METHOD__ . ": user talk page edited, but user does not exist\n" );
-			} elseif ( $targetUser->getId() == $editor->getId() ) {
-				wfDebug( __METHOD__ . ": user edited their own talk page, no notification sent\n" );
-			} elseif ( $targetUser->getOption( 'enotifusertalkpages' ) &&
-				( !$minorEdit || $targetUser->getOption( 'enotifminoredits' ) ) )
-			{
-				if ( $targetUser->isEmailConfirmed() ) {
-					wfDebug( __METHOD__ . ": sending talk page update notification\n" );
-					return true;
-				} else {
-					wfDebug( __METHOD__ . ": talk page owner doesn't have validated email\n" );
-				}
-			} else {
-				wfDebug( __METHOD__ . ": talk page owner doesn't want notifications\n" );
+	private function canSendUserTalkEmail() {
+		$targetUser = User::newFromName( $this->title->getText() );
+		if ( $targetUser instanceof User && // user exists
```

Comment:
True, this could be done as a simple return. Considering how many checks this method makes you juggle in your head, I kind of like the explicit `return true;` and `return false`. For me it just slightly  more clear. I'm willing to sacrifice a couple extra lines for that :)